pkg/apis/core/types_shoot.go

```diff
@@ -355,7 +355,7 @@ type DNS struct {
 
 // DNSProvider contains information about a DNS provider.
 type DNSProvider struct {
-	// TODO(timuthy): Remove this field with release v1.87.
+	// TODO(timuthy): Remove this field after v1.96 has been released.
```

Comment:
```suggestion
	// TODO(timuthy): Remove this field in the scope of https://github.com/gardener/gardener/issues/9176.
```

I created a separate issue for it. We can't quote a concrete release here since it won't be a no-brainer to remove this field. Otherwise we'll fall into the same trap as today.